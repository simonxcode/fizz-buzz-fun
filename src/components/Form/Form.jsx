import React from 'react'
import Result from '../Result';
import useForm from '../../hooks/useForm';

const Form = () => {
  const { values, submissions, errors, handleChange, handleSubmit } = useForm()

  return (
    <div className="p-4 bg-green-50">
      <form onSubmit={handleSubmit} className="p-4  border-solid border bg-gray-50">
        <h1 className="text-2xl font-bold">Signup to join the fun</h1>
        <div>
          <label className="m-2 font-bold">Username:</label>
          <input
            autoFocus={true}  
            type='text'
            name='userName'
            onChange={handleChange}
            value={values.userName || ''}
            className="border-solid border-2 border-current m-2"
          />
          {errors.userName && (<p>{errors.userName}</p>)}
        </div>
        <div>
          <div className="m-2 pr-4">
            <p className="font-bold">Gender:</p>
            <div>
              <label className="ml-4 mr-2">male</label>
              <input
                type='radio'
                name='gender'
                value='male'
                onChange={handleChange}
              />
              <label className="mr-2 ml-4">female</label>
              <input
                type='radio'
                name='gender'
                value='female'
                onChange={handleChange}
              />
              <label className="mr-2 ml-4">other</label>
              <input
                type='radio'
                name='gender'
                value='other'
                onChange={handleChange}
              />
              {errors.gender && (<p>{errors.gender}</p>)}
            </div>
          </div>
          <div className="m-2 pr-4">
            <p className="font-bold">Marital Status:</p>  
            <div>
              <label className="ml-4 mr-2">single</label>
              <input
                type='radio'
                name='maritalStatus'
                value='single'
                onChange={handleChange}
              />
              <label className="ml-4 mr-2">married</label>
              <input
                type='radio'
                name='maritalStatus'
                value='female'
                onChange={handleChange}
              />
              <label className="ml-4 mr-2">other</label>
              <input
                type='radio'
                name='maritalStatus'
                value='other'
                onChange={handleChange}
              />
              {errors.maritalStatus && (<p>{errors.maritalStatus}</p>)}
            </div>
          </div>
          <div>
            <label className="m-2 font-bold">Date of Birth:</label>
            <input
              type='date'
              name='dateOfBirth'
              onChange={handleChange}
              value={values.dateOfBirth || ''}
              className="border-solid border-2 border-current m-2"
            />
            {errors.dateOfBirth && (<p>{errors.dateOfBirth}</p>)}
          </div>
          <div>
            <label className="m-2 font-bold">Lucky Number:</label>
            <input
              type='number'
              name='luckyNumber'
              onChange={handleChange}
              value={values.luckyNumber || ''}
              className="border-solid border-2 border-current m-2"
            />
            {errors.luckyNumber && (<p>{errors.luckyNumber}</p>)}
          </div>
        </div>
        <button type='submit'className="border-solid border-2  border-current m-2 pl-2 pr-2 rounded-md">
          Submit
        </button>
      </form>
      {submissions.map((submission, index) => (
        <Result key={index} submission={submission} />
        ))}
    </div>
  )
}

export default Form;
