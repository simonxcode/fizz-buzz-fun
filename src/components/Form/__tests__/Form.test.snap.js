import React from 'react'
import renderer from 'react-test-renderer'
import Form from '../Form'

it('Form component should match snapshot', () => {
  const tree = renderer.create( < Form / > ).toJSON()
  expect(tree).toMatchSnapshot()
});
