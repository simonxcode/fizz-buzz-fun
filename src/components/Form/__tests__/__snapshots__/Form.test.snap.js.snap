// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should match snapshot 1`] = `
<div
  className="p-4 bg-green-50"
>
  <form
    className="p-4  border-solid border bg-gray-50"
    onSubmit={[Function]}
  >
    <h1
      className="text-2xl font-bold"
    >
      Signup to join the fun
    </h1>
    <div>
      <label
        className="m-2 font-bold"
      >
        Username:
      </label>
      <input
        autoFocus={true}
        className="border-solid border-2 border-current m-2"
        name="userName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <div
        className="m-2 pr-4"
      >
        <p
          className="font-bold"
        >
          Gender:
        </p>
        <div>
          <label
            className="ml-4 mr-2"
          >
            male
          </label>
          <input
            name="gender"
            onChange={[Function]}
            type="radio"
            value="male"
          />
          <label
            className="mr-2 ml-4"
          >
            female
          </label>
          <input
            name="gender"
            onChange={[Function]}
            type="radio"
            value="female"
          />
          <label
            className="mr-2 ml-4"
          >
            other
          </label>
          <input
            name="gender"
            onChange={[Function]}
            type="radio"
            value="other"
          />
        </div>
      </div>
      <div
        className="m-2 pr-4"
      >
        <p
          className="font-bold"
        >
          Marital Status:
        </p>
        <div>
          <label
            className="ml-4 mr-2"
          >
            single
          </label>
          <input
            name="maritalStatus"
            onChange={[Function]}
            type="radio"
            value="single"
          />
          <label
            className="ml-4 mr-2"
          >
            married
          </label>
          <input
            name="maritalStatus"
            onChange={[Function]}
            type="radio"
            value="female"
          />
          <label
            className="ml-4 mr-2"
          >
            other
          </label>
          <input
            name="maritalStatus"
            onChange={[Function]}
            type="radio"
            value="other"
          />
        </div>
      </div>
      <div>
        <label
          className="m-2 font-bold"
        >
          Date of Birth:
        </label>
        <input
          className="border-solid border-2 border-current m-2"
          name="dateOfBirth"
          onChange={[Function]}
          type="date"
          value=""
        />
      </div>
      <div>
        <label
          className="m-2 font-bold"
        >
          Lucky Number:
        </label>
        <input
          className="border-solid border-2 border-current m-2"
          name="luckyNumber"
          onChange={[Function]}
          type="number"
          value=""
        />
      </div>
    </div>
    <button
      className="border-solid border-2  border-current m-2 pl-2 pr-2 rounded-md"
      type="submit"
    >
      Submit
    </button>
  </form>
  <div
    className="p-4 bg-gray-200"
  >
    <div
      className="bg-gray-50 p-4 border-solid border"
    >
      <div
        className="flex"
      >
        <p
          className="font-bold pr-2"
        >
          Username:
        </p>
        <p>
          april123
        </p>
      </div>
      <div
        className="flex"
      >
        <p
          className="font-bold pr-2"
        >
          Gender:
        </p>
        <p>
          female
        </p>
      </div>
      <div
        className="flex"
      >
        <p
          className="font-bold pr-2"
        >
          Date of Birth:
        </p>
        <p>
          2000-01-01
        </p>
      </div>
      <div
        className="flex"
      >
        <p
          className="font-bold pr-2"
        >
          Lucky Number:
        </p>
        <p>
          3
        </p>
      </div>
      <div
        className="flex"
      >
        <p
          className="font-bold pr-2"
        >
          Team:
        </p>
        <p>
          fizz
        </p>
      </div>
    </div>
  </div>
</div>
`;
